apollo-cache-interceptor/src/commonMain/kotlin/com/apollographql/apollo/interceptor/cache/ApolloCacheInterceptor.kt

```diff
@@ -0,0 +1,95 @@
+package com.apollographql.apollo.interceptor.cache
+
+import com.apollographql.apollo.api.ApolloExperimental
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.api.Response.Companion.builder
+import com.apollographql.apollo.api.ResponseField
+import com.apollographql.apollo.api.internal.NoOpResolveDelegate
+import com.apollographql.apollo.cache.CacheHeaders
+import com.apollographql.apollo.cache.normalized.CacheKey
+import com.apollographql.apollo.cache.normalized.CacheKeyResolver
+import com.apollographql.apollo.cache.normalized.Record
+import com.apollographql.apollo.cache.normalized.internal.CacheFieldValueResolver
+import com.apollographql.apollo.cache.normalized.internal.CacheKeyBuilder
+import com.apollographql.apollo.cache.normalized.internal.ReadableStore
+import com.apollographql.apollo.cache.normalized.internal.RealCacheKeyBuilder
+import com.apollographql.apollo.cache.normalized.internal.ResponseNormalizer
+import com.apollographql.apollo.cache.normalized.simple.SimpleNormalizedCache
+import com.apollographql.apollo.interceptor.ApolloInterceptorChain
+import com.apollographql.apollo.interceptor.ApolloRequest
+import com.apollographql.apollo.interceptor.ApolloRequestInterceptor
+import com.apollographql.apollo.interceptor.ApolloResponse
+import com.apollographql.apollo.internal.response.RealResponseReader
+import com.apollographql.apollo.internal.response.RealResponseWriter
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+
+@ApolloExperimental
+class ApolloCacheInterceptor : ApolloRequestInterceptor {
+  val normalizedCache = SimpleNormalizedCache()
+  val readableStore = object : ReadableStore {
```

Comment:
As far as I remember the original idea is that interceptor get `*Store` implementation injected where `*Store` is constructed with cache implementation. So interceptor knows only about store, and has only one dependency.

Reply:
Makes perfect sense, I changed that.